libs/platform/experience/carousel-navigation/carousel-navigation.component.ts

```diff
@@ -0,0 +1,410 @@
+import { IconTypes } from '@spryker-oryx/ui/icon';
+import { throttle } from '@spryker-oryx/utilities';
+import { LitElement, PropertyValueMap, TemplateResult, html } from 'lit';
+import { property, queryAll, state } from 'lit/decorators.js';
+import {
+  ArrowNavigationBehavior,
+  CarouselIndicatorAlignment,
+  CarouselIndicatorPosition,
+  CarouselLayoutProperties,
+  CarouselScrollBehavior,
+} from '../src/services';
+import { carouselNavigationStyles } from './carousel-navigation.styles';
+
+export class CarouselNavigationComponent
+  extends LitElement
+  implements CarouselLayoutProperties
+{
+  static styles = carouselNavigationStyles;
+
+  @property({ type: Boolean }) showArrows?: boolean;
+  @property({ type: Boolean }) showIndicators?: boolean;
+  @property() arrowNavigationBehavior?: ArrowNavigationBehavior;
+  @property({ reflect: true }) indicatorsPosition?: CarouselIndicatorPosition;
+  @property({ reflect: true }) indicatorsAlignment?: CarouselIndicatorAlignment;
+  @property() scrollBehavior?: CarouselScrollBehavior;
+
+  protected resizeObserver?: ResizeObserver;
+  protected intersectionObserver?: IntersectionObserver;
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  protected throttledScrollListener: () => void = () => {};
+
+  @state() protected items: HTMLElement[] = [];
+  @state() protected slides: { index: number }[] = [];
+
+  @queryAll('input') protected indicatorElements?: NodeListOf<HTMLInputElement>;
+
+  /**
+   * @override Initialize the intersection observer. The intersection observer is
+   * used to detect when the carousel becomes visible and sets the resize observer
+   * and listens to scroll events to always keep the state up to date.
+   */
+  connectedCallback(): void {
+    this.initializeIntersectionObserver();
+    super.connectedCallback();
+  }
+
+  disconnectedCallback(): void {
+    this.resizeObserver?.disconnect();
+    this.intersectionObserver?.disconnect();
+    if (this.hostElement) {
+      this.hostElement.removeEventListener(
+        'scroll',
+        this.throttledScrollListener
+      );
+      this.hostElement.removeEventListener('scrollend', this.updateState);
+    }
+    super.disconnectedCallback();
+  }
+
+  /**
+   * Initializes the IntersectionObserver to detect when the carousel becomes visible.
+   * This is necessary to calculate the carousel's width and to build the navigation.
+   */
+  protected initializeIntersectionObserver(): void {
+    if (this.intersectionObserver) return;
+
+    this.intersectionObserver = new IntersectionObserver(
+      throttle((entries: IntersectionObserverEntry[]) => {
+        return entries.forEach((entry) => {
```

Comment:
I didn't get, why do you need to cal `buildNavigation`,  `initializeResizeObserver` and the rest inside foreach? It will call this actions multiple times regarding the amount of entries

Reply:
That's how the intersection observer works. I've done a small optimisation to ensure that we only initialise once.